File: app/src/main/java/com/gmail/jaaska/jaakko/calendarcountdown/ui/CountdownsRecyclerViewAdapter.java
package com.gmail.jaaska.jaakko.calendarcountdown.ui;

import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.gmail.jaaska.jaakko.calendarcountdown.data.CountdownSettings;
import com.gmail.jaaska.jaakko.calendarcountdown.R;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

/**
 * Created by jaakko on 16.5.2016.
 */
public class CountdownsRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<CountdownSettings> items;
    private static final String TAG = "CntdwnsRcclrVwAdap";

    public CountdownsRecyclerViewAdapter(List<CountdownSettings> items) {
        this.items = items;
    }


    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.listitem_countdown, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        CountdownSettings item = items.get(position);
        ViewHolder vh = (ViewHolder) holder;

        String dateString = new SimpleDateFormat("d.M.yyyy").format(new Date(item.getEndDate()));
        vh.textViewEndDate.setText(dateString);
        vh.textViewDaysCount.setText(Integer.toString(item.getDaysToEndDate()));
        vh.textViewLabel.setText(item.getLabel());
        vh.item = item;
    }

    @Override
    public int getItemCount() {
        Log.d(TAG, "getItemCount() - items count: "+Integer.toString(items.size()));
        return items.size();
    }

    private static class ViewHolder extends RecyclerView.ViewHolder {

        public TextView textViewDaysCount;
        public TextView textViewLabel;
        public TextView textViewEndDate;

        public CountdownSettings item;

        public ViewHolder(View itemView) {
            super(itemView);

            textViewDaysCount = (TextView) itemView.findViewById(R.id.textViewCountdownDaysToGo);
            textViewLabel = (TextView) itemView.findViewById(R.id.textViewCountdownLabel);
            textViewEndDate = (TextView) itemView.findViewById(R.id.textViewCountdownEndDate);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent setupIntent = new Intent(v.getContext(), SetupActivity.class);
                    setupIntent.putExtra(CountdownSettings.extraName, item);
                    v.getContext().startActivity(setupIntent);
                }
            });

        }
    }
}

File: app/src/main/java/zak0/github/calendarcountdown/ui/CountdownsRecyclerViewAdapter.kt
package zak0.github.calendarcountdown.ui

import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import zak0.github.calendarcountdown.R
import zak0.github.calendarcountdown.data.CountdownSettings
import zak0.github.calendarcountdown.util.DateUtil
import kotlinx.android.synthetic.main.listitem_countdown.view.*

/**
 * Created by jaakko on 16.5.2016.
 */
class CountdownsRecyclerViewAdapter(private val items: List<CountdownSettings>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.listitem_countdown, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val item = items[position]
        val vh = holder as ViewHolder

        vh.textViewEndDate.text = DateUtil.databaseDateToUiDate(item.endDate)
        vh.textViewDaysCount.text = "${item.daysToEndDate}"
        vh.textViewLabel.text = item.label
        vh.item = item
    }

    override fun getItemCount(): Int {
        Log.d(TAG, "getItemCount() - items count: " + Integer.toString(items.size))
        return items.size
    }

    private class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var textViewDaysCount: TextView = itemView.textViewCountdownDaysToGo
        var textViewLabel: TextView = itemView.textViewCountdownLabel
        var textViewEndDate: TextView = itemView.textViewCountdownEndDate

        var item: CountdownSettings? = null

        init {
            itemView.setOnClickListener { v ->
                val setupIntent = Intent(v.context, SetupActivity::class.java)
                setupIntent.putExtra(CountdownSettings.extraName, item)
                v.context.startActivity(setupIntent)
            }

        }
    }

    companion object {
        private val TAG = CountdownsRecyclerViewAdapter::class.java.simpleName
    }
}
